### ISSUE TITLE
Cannot create container when using a secure connection to docker

### ISSUE THREAD

**Issue by myers** [swipely/docker-api#214]
```ruby
require 'docker'

c = Docker::Container.create('Cmd' => ['ls'], 'Image' => 'code-challenge-runner')
c.start
# this line causes the following error
c.attach(stdin: StringIO.new(%Q|ruby -e 'puts "sleeping"; sleep(60)'\n|))
```

```
/Users/myers/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/docker-api-1.15.0/lib/docker/container.rb:217:in `block (2 levels) in hijack_for': undefined method `close_write' for #<OpenSSL::SSL::SSLSocket:0x007fad82d3cc28> (NoMethodError) (Excon::Errors::SocketError)
```

Indeed [OpenSSL::SSLSocket](http://ruby-doc.org/stdlib-2.1.5/libdoc/openssl/rdoc/OpenSSL/SSL/SSLSocket.html) does not have the close_write method.

Using boot2docker VM to deploy against.  It by default uses a secure connection.

**Comment by dividedmind** [comment]
Hi, I've had success fixing this by opening a separate connection for stdin, cf. https://github.com/conjurinc/buncker/commit/ab820b5c610ec2a292ad4d5b588f18927ae00a05

**Comment by tlunter** [comment]
This has been released with closing the IO of OpenSSL::SSLSocket which worked for us. https://github.com/swipely/docker-api/blob/master/lib/docker/util.rb#L87-L95

**Comment by spraints** [comment]
For future reference, this was fixed in #241 and was first released in 1.19.0.
